roles/cloudscale_server/defaults/main.yml

```diff
@@ -0,0 +1,39 @@
+---
+cloudscale_server__name: "{{ inventory_hostname_short }}"
+cloudscale_server__image: debian-10
+cloudscale_server__flavor: flex-2
+cloudscale_server__force: false
+cloudscale_server__use_public_network: true
+cloudscale_server__use_private_network: false
+cloudscale_server__use_ipv6: true
+cloudscale_server__root_volume_size: 10
+
+cloudscale_server__interfaces: []
+# - network: public
+# - network: 2db69ba3-1864-4608-853a-0771b6885a3a
+
+cloudscale_server__volumes: []
+# Note: name will be prefixed with vm name by default
+# - name: ssd1
+#   type: ssd,
+#   size_gb: 10,
+#   state: detached
+# - name: bulk1
+#   type: bulk
+#   size_gb: 100
+
+cloudscale_server__server_groups: []
+cloudscale_server__password: "{{ omit }}"
+cloudscale_server__ssh_pubkey_path: "~/.ssh/id_ed25519.pub"
```

Comment:
This doesn't exist in my setup, which prompted me to think if maybe this could be moved to be alongside the token (though secrets.yml would no longer be a good name). I would think that this is something that needs to be configured often, so putting it into a file the user has to edit anyway might be a good idea.

Reply:
There are already various options and ways to set a variable:

* by group_/host_vars
* extra_vars... 
* include_var in a playbook 
* vars in a playbook
* extend the role by an own role and redefine the defaults
* include_role with vars

One could set a `group_vars/all.yml` with `cloudscale_server__ssh_pubkey_path: "{{ lookup('env', 'SSH_KEY_PATH') }}"` or even use the `cloudscale_server__ssh_keys` to query the keys from github. this is really an implementation detail. 

This is really just a default of the variable and I found it better to default it to something then just leave it empty. It does not mean to work for everyone.

Reply:
Though certainly true that this can be done in many places, I figured this would be one configuration item that needs to change as part of the course of the example, so I would have preferred a designated area for this. My worry is that this is used by someone that is new to Ansible and then ends up being a support case on our end.

However, its not a major point for me, so I'm okay with leaving it as is. This can always be added if it indeed becomes something we are asked about.